js/src/modal.js

```diff
@@ -346,18 +344,16 @@ class Modal extends BaseComponent {
     }
 
     this._element.classList.add(CLASS_NAME_STATIC)
-    const modalTransitionDuration = getTransitionDurationFromElement(this._dialog)
-    EventHandler.off(this._element, 'transitionend')
-    EventHandler.one(this._element, 'transitionend', () => {
+
+    this._queueCallback(() => {
       this._element.classList.remove(CLASS_NAME_STATIC)
       if (!isModalOverflowing) {
-        EventHandler.one(this._element, 'transitionend', () => {
+        this._queueCallback(() => {
           this._element.style.overflowY = ''
-        })
-        emulateTransitionEnd(this._element, modalTransitionDuration)
+        }, this._dialog)
       }
-    })
-    emulateTransitionEnd(this._element, modalTransitionDuration)
+    }, this._dialog)
+
```

Comment:
This used the duration from `this._dialog` before, but added the event listener to `this._element`.

This was somewhat working but prone to errors, because it ignored the fact that `this._element` could dispatch `transitionend` events itself (or have listeners added that would have been called by this code).

Reply:
Now the handler for `transitionend` event will not be removed before handling the callback?
```diff
-EventHandler.off(this._element, 'transitionend')
```

Reply:
As now listens only to `_dialog` seems to be ok. Before it was listening to  `_element` event. So it had to de-register other listeners, to avoid being triggered

Reply:
According to https://github.com/twbs/bootstrap/pull/30326#discussion_r443898058 this was added because scrollbars still showed up when clicking multiple times. I couldn't reproduce that behaviour when I removed that line. I compared the before and after behaviour and it worked the same.

I agree that this isn't ideal because it causes those callbacks to be called all at once, but since there is only one transitionend event dispatched by the browser it shouldn't be really a problem.

If you want me to clear that so we make sure the callback is only called once I can readd this.

Reply:
@alpadev in this PR or later? (merge it or not?)  



PS: please squash your commits ;)

Reply:
@GeoSot squashed it

@rohit2sharma95 added some code to prevent `_queueCallback` to be called multiple times. I'd like to mention that the implementation before was removing the handlers, but there were several `emulateTransitionEnd` callbacks queued. This is prevented now as well. Would like to ask you, to have a second look.